fiftyone/core/view.py

```diff
@@ -1331,6 +1341,7 @@ def _add_view_stage(self, stage):
             media_type = stage.get_media_type(self)
             if media_type is not None:
                 view._set_media_type(media_type)
+            view._set_name(None)
```

Comment:
This needs to be outside the `if` block. The `stage.has_view` case also returns a different view which is not the saved view.